src/Curl/Curl.php

```diff
@@ -232,7 +232,19 @@ protected function preparePayload($data)
         $this->setOpt(CURLOPT_POST, true);
 
         if (is_array($data) || is_object($data)) {
-            $data = http_build_query($data);
+            $skip = false;
+            foreach ($data as $key => $value) {
+                // If a value is an instance of CurlFile skip the http_build_query
+                // see issue https://github.com/php-mod/curl/issues/46
+                // suggestion from: https://stackoverflow.com/a/36603038/4611030
+                if ($value instanceof \CurlFile) {
+                    $skip = true;
+                }
+            }
+            
+            if (!$skip) {
+                $data = http_build_query($data);
```

Comment:
We cannot skip http_build_query if there is a CurlFile, we should work arround this by using something like: https://stackoverflow.com/a/35741943

Reply:
I tested on local application, it works fine, at least it's sending expected content to the API when skipping http_build_query as long as CurlFile is available. If you have another solution, its fine for me as well. This is a very urgent issue for your headless library.